ds-collector/ds-collector

```diff
@@ -122,19 +122,19 @@ list_cassandra_nodes() {
   ( timeout --help | grep -q "t SECS" ) && TIMEOUT_OPT="-t 3"
   timeout_command="timeout $FOREGROUND_OPT $TIMEOUT_OPT"
 
-  nodetoolHost=''
+  jmxHost='127.0.0.1'
   $timeout_command nc -zv localhost $jmxPort > /dev/null 2>&1
   if [ $? = 0 ]; then
-    nodetoolHost='-h localhost'
+    jmxHost='localhost'
     echo "Using localhost to connect to JMX..."
   else
     $timeout_command nc -zv $(hostname) $jmxPort > /dev/null 2>&1
     if [ $? = 0 ]; then
-      nodetoolHost="-h $(hostname)"
+      jmxHost="$(hostname)"
       echo "Using $(hostname) to connect to JMX..."
     fi
   fi
-  cmd="set -o pipefail ; ${nodetoolCmd} $nodetoolHost -p $jmxPort ${jmxOpts} $nodetoolCredentials status | grep UN | tr -s ' ' | cut -d' ' -f2"
+  cmd="set -o pipefail ; ${nodetoolCmd} -h $jmxHost -p $jmxPort ${jmxOpts} $nodetoolCredentials status | grep UN | tr -s ' ' | cut -d' ' -f2"
```

Comment:
this is not the change, as you're just changing a variable name.  But there is no test/assertion that `jmxHost` is set...i.e. it is possible there is nothing listening on the port, at which point the `cmd` will not be happy in some way, and at a point somewhere else in the code.

Reply:
we do definitely need to put more time and effort into a) fail-fast informative error messages (with the goal of helping the user configure collector.conf correctly as quickly as possible), and b) a bash style standard enforced.